botocore/signers.py

```diff
@@ -127,6 +117,36 @@ def sign(self, operation_name, request):
                                             signature_version)
             signer.add_auth(request=request)
 
+    def choose_signer(self, operation_name, suffix=''):
```

Comment:
I was first hesitant about exposing this as a public method. But it looks like the ``generate_presigned_post`` uses it, and I think it is a reasonable workflow for people want to use the signer to implement their own auth to call:

1) call choose_signer
2) call get_auth_instance with the retrieved signature version
3) sign the request with the auth_instance

So I am fine with adding this.